eth/tracers/live/firehose.go

```diff
@@ -825,10 +825,6 @@ func (f *Firehose) OnNewAccount(a libcommon.Address) {
 		return
 	}
 
-	if f.isPrecompileAddress(a) {
-		return
-	}
-
```

Comment:
How this method is not working but propagating the precompile down to StateDB works. I think it will be important to understand what is the difference.

That being said, the native tracer Git commit https://github.com/s1na/go-ethereum/commit/ee58cc79d9dd4631e87cad30338543324434f20d implements it differently, I think porting those will be needed

Reply:
> That being said, the native tracer Git commit https://github.com/s1na/go-ethereum/commit/ee58cc79d9dd4631e87cad30338543324434f20d implements it differently, I think porting those will be needed

These changes are already ported in [statedb: precompile check before onNewAccount](https://github.com/graphprotocol/erigon/pull/52/commits/5d64c1d44611f04e736fe8392531cb2857e3cc02) commit. I removed precompile check from here since it being done at statedb now.

> How this method is not working but propagating the precompile down to StateDB works. I think it will be important to understand what is the difference.

In the old firehose implementation, we take the precompile flag on the AddBalance [here](https://github.com/streamingfast/go-ethereum/blob/firehose-fh2.3/core/state/statedb.go#L393). If you check the references of `AddBalance`, In majority of the cases we are sending hardcoded `false` instead of checking against the precompile config. Because of this even precompiled contracts are getting logged in the `NewAccount` in some cases.
Whereas in new implementation, each and every account was checked against the precompile config.

In this commit,  I added a new flag `checkPrecompile`, that denotes if new account should be checked against precompiled address or not. I am sending `checkPrecompile` values according to the old implementation to make the new tracer backward compatible.

@maoueh Let me know what you think